src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -544,9 +604,18 @@ private static void handleKeyValueConfig(TestInputConfiguration.Builder testInpu
             final ModuleInputConfiguration.Builder moduleInputConfigBuilder =
                     new ModuleInputConfiguration.Builder();
             final String moduleName = lines.get(lineNo);
+            Map<String, String> defaultProperties = new HashMap<>();
+            if (!moduleName.contains(".") && !SUPPRESSED_MODULES.contains(moduleName)) {
+                defaultProperties = getPropertiesWithDefaults(
+                    getMetaPath(inputFilePath, moduleName),
+                    inputFilePath);
+            }
             setModuleName(moduleInputConfigBuilder, inputFilePath, moduleName);
-            setProperties(moduleInputConfigBuilder, inputFilePath, lines, lineNo + 1, moduleName);
+            validateDefaultProperties(lines, lineNo + 1, defaultProperties);
```

Comment:
```suggestion
            validateDefaultProperties(loadProperties(readPropertiesContent(lines, lineNo + 1)), defaultProperties);
```
reduce coupling by always giving method what it actually needs to avoid feature envy.

Reply:
done.